hal/src/thumbv7em/watchdog.rs

```diff
@@ -30,24 +30,60 @@ impl Watchdog {
     }
 }
 
+fn feed(&mut wdt: WDT) {
```

Comment:
This could be the way to go, until e-h traits aren't "unproven" anymore. But I am open for suggestions/better options.